### DISCUSSION TITLE
OCP for heat source calculation

### DISCUSSION THREAD

**Discussion by xyyrly** [pybamm-team/PyBaMM#3088] [Q&A] [answered]
Hi Team~

I guess it is a basic question, but how does Pybamm calculate open circult potential U and dU/dT when calculating reversiable and irreverisble heating source? Is there somewhere I can check the formula U(c_s/c_max,T)?  

Cannot thank you more!

**Comment by ben-kenney** [comment, marked as answer]
Those can be found in the parameter set that you are using. For example, in [Mohtat2020](https://github.com/pybamm-team/PyBaMM/blob/5e28bb9a316f394282b40c8a56b285e85d987b77/pybamm/input/parameters/lithium_ion/Mohtat2020.py) you'll find functions for:

graphite_ocp_PeymanMPM(sto)
graphite_entropic_change_PeymanMPM(sto, c_s_max)

NMC_ocp_PeymanMPM(sto)
NMC_entropic_change_PeymanMPM(sto, c_s_max)

All parameter sets would have the OCV as a function of stoichiometry but not all parameter sets would have dU/dT unless they support the thermal options.

**Reply by xyyrly** [reply]
Thank you veryyyy much, that really helps! I found one from [Ai2020.py](https://github.com/pybamm-team/PyBaMM/blob/5e28bb9a316f394282b40c8a56b285e85d987b77/pybamm/input/parameters/lithium_ion/Ai2020.py)
